internal/model/rwfolder.go

```diff
@@ -1222,8 +1222,44 @@ func (p *rwFolder) performFinish(state *sharedPullerState) error {
 		}
 	}
 
-	// Set the correct timestamp on the new file
+	// Verify if a file will be overwritten and if we are up to date about its current mtime
 	t := time.Unix(state.file.Modified, 0)
+	// This call is really expensive for large files
+	p.model.fmut.RLock()
+	lf, ok := p.model.folderFiles[p.folder].Get(protocol.LocalDeviceID, state.file.Name)
```

Comment:
Sorry, can't follow. Is it to optimise the code via folderRunners (if possible didn't check) or is it a bug?